### PR TITLE
Move pay button to the footer

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-react/src/core/utils/isInputAmountUsed.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {TopUpProvider} from '../models/TopUpProvider';
+
+export const isInputAmountUsed = (topUpProvider?: TopUpProvider) => {
```
Comment (Marik-D) [inline universal-login-react/src/core/utils/isInputAmountUsed.ts]: Can you remove it to something like: `providerUsesInputAmount` or `providerAllowsToPrefillAmount` or `amountPrefillableForProvider`

File: universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx [hunk 2]
```diff
@@ -71,22 +80,36 @@ export const ChooseTopUpMethod = ({sdk, contractAddress, onPayClick, topUpClassN
...
                 isDeployed={false}
                 sdk={sdk}
                 onGasParametersChanged={onGasParametersChanged!}
                 gasLimit={DEPLOY_GAS_LIMIT}
                 className={topUpClassName}
               />
-            </FooterSection>
-          }
+            }
+            <button
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx]: Extract this button as for example `PayButton`

File: universal-login-react/src/core/models/FiatOptions.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import {TopUpProvider} from './TopUpProvider';
+
+export interface FiatOptions {
+  topUpProvider?: TopUpProvider;
+  amount: string;
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/core/models/FiatOptions.ts]: Make amount optional

File: universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import {TopUpProvider} from '../models/TopUpProvider';
+
+const providersUsedPrefilledAmount = [TopUpProvider.RAMP];
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts]: `providersRequiresAmount`

File: universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+export const isPrefilledAmountUsedBy = (topUpProvider?: TopUpProvider) => {
```
Comment (Marik-D) [inline universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts]: Why did you move it out of the service, that seemed like a better place for this function?

File: universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts (hunk 4 above)
Comment (cehali) [reply universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts]: I understood that the provider does not require amount but it is possible to pass the amount. `providersUsedPrefilledAmount` is not the best name also.

File: universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts (hunk 5 above)
Comment (cehali) [reply universal-login-react/src/core/utils/isPrefilledAmountUsedBy.ts]: 👍

**Comment by Marik-D** [comment]
Close for now
